pkg/controller/servicecatalog/provisionedproduct/utils.go

```diff
@@ -0,0 +1,101 @@
+package provisionedproduct
+
+import (
+	"fmt"
+
+	cfsdkv2types "github.com/aws/aws-sdk-go-v2/service/cloudformation/types"
+	svcsdk "github.com/aws/aws-sdk-go/service/servicecatalog"
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	"k8s.io/utils/pointer"
+
+	svcapitypes "github.com/crossplane-contrib/provider-aws/apis/servicecatalog/v1alpha1"
+	aws "github.com/crossplane-contrib/provider-aws/pkg/clients"
+)
+
+func provisioningParamsAreChanged(cfStackParams []cfsdkv2types.Parameter, currentParams []*svcapitypes.ProvisioningParameter) bool {
+	if len(cfStackParams) != len(currentParams) {
+		return true
+	}
+
+	cfStackKeyValue := make(map[string]string)
+	for _, v := range cfStackParams {
+		cfStackKeyValue[*v.ParameterKey] = pointer.StringDeref(v.ParameterValue, "")
+	}
+
+	for _, v := range currentParams {
+		if cfv, ok := cfStackKeyValue[*v.Key]; ok && pointer.StringEqual(&cfv, v.Value) {
+			continue
+		} else {
+			return true
+		}
+	}
+
+	return false
+}
+
+// productOrArtifactAreChanged will attempt to determine whether or not the currently requested SC product and provisioning artifact IDs have changed when using names instead of IDs
+func (c *custom) productOrArtifactAreChanged(ds *svcapitypes.ProvisionedProductParameters, resp *svcsdk.ProvisionedProductDetail) (bool, error) { // nolint:gocyclo
```

Comment:
`// nolint:gocyclo` I think the idea of this rule is to keep code clean and protect it from some things... : )